### PR TITLE
Fix error handling when go command is missing

### LINKED ISSUE #38070: download-frozen-image-v2.sh does not contain a check to detect if go is in the PATH
**Description**

download-frozen-image-v2.sh does not contain a check to detect if go is in the PATH. It checks for curl and jq, it should do the same for go.

**Steps to reproduce the issue:**

1. Run the script on a machine where go is not installed, you will hit the following error:

```
./download-frozen-image-v2.sh /tmp/old-hello-world hello-world:latest
./download-frozen-image-v2.sh: line 41: go: command not found
```

**Describe the results you expected:**

I expect the script to throw an error like it does for jq:

```
$ ./download-frozen-image-v2.sh /tmp/old-hello-world hello-world:latest
error: "go" not found!
```

### PR DESCRIPTION
**Pull request by mjeromin**
Signed-off-by: Mark Jeromin <mark.jeromin@sysfrog.net>

Fixes #38070 **_download-frozen-image-v2.sh does not contain a check to detect if go is in the PATH_**

**- What I did**
I added a test to check that `go` is in the user's shell `$PATH`, otherwise it will print a friendly error message `error: "go" not found!` when `go` is not in the `$PATH`. 

**- How I did it**
@thaJeztah suggested a solution that doesn't increase the complexity of the script much, so I consolidated the tests into a single loop. This allows a consistent error message for missing commands, rather than three (or more) separate tests (ie. `curl`, `jq`, `go`).

**- Description for the changelog**
Refactor error handling for missing commands.

jack 016-xl

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: contrib/download-frozen-image-v2.sh [hunk 1]
```diff
@@ -6,14 +6,14 @@ set -eo pipefail
...
-if ! command -v curl &> /dev/null; then
-	echo >&2 'error: "curl" not found!'
-	exit 1
-fi
-if ! command -v jq &> /dev/null; then
-	echo >&2 'error: "jq" not found!'
-	exit 1
-fi
+
+# Test essential commands are in our PATH
```
Comment (thaJeztah) [inline contrib/download-frozen-image-v2.sh]: Nit; I see other comments in this file start with a lowercase; for consistency, perhaps change this comment as well. Maybe something like;

```suggestion
# check if essential commands are in our PATH
```

File: contrib/download-frozen-image-v2.sh [hunk 2]
```diff
@@ -6,14 +6,14 @@ set -eo pipefail
...
+for cmd in curl jq go; do
+    if ! command -v $cmd &> /dev/null; then
+	    echo >&2 "error: \"$cmd\" not found!"
```
Comment (thaJeztah) [inline contrib/download-frozen-image-v2.sh]: oh! Looks like you're using mixed tabs and spaces for indentation here 😅 - we standardise on using tabs for indentation in scripts;

```suggestion
# check if essential commands are in our PATH
for cmd in curl jq go; do
	if ! command -v $cmd &> /dev/null; then
		echo >&2 "error: \"$cmd\" not found!"
		exit 1
	fi
done
```

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Thanks for working on this! I noticed some minor issues; could you amend your commit and update the PR? Let me know if you need help doing so 👍

File: contrib/download-frozen-image-v2.sh (hunk 2 above)
Comment (mjeromin) [reply contrib/download-frozen-image-v2.sh]: Thanks @thaJeztah. The comment and spacing should be consistent now.

**Comment by thaJeztah** [comment]
Thanks! I see you made the changes in a new commit; would you be able to "squash" the commits? You can fine some instructions here; https://docs.docker.com/v17.06/opensource/workflow/work-issue/#pull-and-rebase-frequently

If your upstream remote (the remote referring to github.com/moby/moby) is named "upstream", the steps should be something like this;

```
# switch to your branch if you were not yet in that branch
git checkout 38070-go-command-detect

# make sure your "upstream" remote is up to date (important :smile:)
git fetch upstream -v

# start an interactive rebase
git rebase -i upstream/master
```

That step should show open an editor, showing the commits you made;

```
pick a291343782 Fix error handling when go command is missing
pick 6bd7d87eaa Fixed comments and spacing for consistency

# Rebase 104cbc0780..6bd7d87eaa onto 104cbc0780 (2 commands)
#
```

Change the second commit from "pick" to "squash", and save the file

```
pick a291343782 Fix error handling when go command is missing
squash 6bd7d87eaa Fixed comments and spacing for consistency

# Rebase 104cbc0780..6bd7d87eaa onto 104cbc0780 (2 commands)
#
```


A new editor is opened to edit the commit message for the squashed commit; you can edit that to your liking (be sure that it still has a sign-off)

```
# This is a combination of 2 commits.
# This is the 1st commit message:

Fix error handling when go command is missing

Signed-off-by: Mark Jeromin <mark.jeromin@sysfrog.net>

# This is the commit message #2:

Fixed comments and spacing for consistency

Signed-off-by: Mark Jeromin <mark.jeromin@sysfrog.net>
```


Safe the file (that will complete the rebase)


Now _force push_ your changes; that will update the pull request;

```
git push -f origin     
```

Celebrate 🎉 


Hope that helps, but feel free to ping me if you need more help 🤗

**Review by thaJeztah — COMMENTED** [review]
LGTM after squashing 👍

**Comment by mjeromin** [comment]
I hope that squashed right. Thanks! 🎉

**Review by thaJeztah — APPROVED** [review]
perfect, thank you so much!

LGTM

**Pull request merged**